libraries/datasource/auth/src/main/java/com/chesire/nekome/datasource/auth/local/LocalAuthV2.kt

```diff
@@ -0,0 +1,91 @@
+package com.chesire.nekome.datasource.auth.local
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Build
+import androidx.core.content.edit
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKeys
+import dagger.Reusable
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+private const val ENCRYPTED_NEKOME_AUTH_PREF = "ENekomePrefStore"
+private const val UNENCRYPTED_NEKOME_AUTH_PREF = "UneNekomePrefStore"
+private const val ACCESS_TOKEN = "KEY_KITSU_ACCESS_TOKEN_V2"
+private const val REFRESH_TOKEN = "KEY_KITSU_REFRESH_TOKEN_v2"
+
+@Reusable
+class LocalAuthV2 @Inject constructor(
+    @ApplicationContext private val context: Context
+) : LocalAuth {
+
+    private val preferences: SharedPreferences
+
+    init {
+        val unencryptedPreferences = initializeUnencryptedPreferences()
```

Comment:
Move this to each branch.
Check the encrypted preferences to see if there is a value first, if these is we don't need to construct this at all since we don't need to migrate anything from pref>encryptedpref